### PR TITLE
SDFSOF-64-PR2: RPC Actions: implement logic and validations

### PR DESCRIPTION
**Pull request by MazurakIhor**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

RPC actions and unit tests:

- do_stellar_payment
- notify_refund_initiated
- notify_onchain_funds_sent
- notify_trust_set
- request_trust

### Why

RPC actions should replace PATCH /transactions endpoint

### CODE REVIEW

File: platform/src/main/java/org/stellar/anchor/platform/action/NotifyOnchainFundsSentHandler.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  @Override
+  public ActionMethod getActionType() {
+    return NOTIFY_ONCHAIN_FUNDS_SENT;
+  }
+
+  @Override
+  protected SepTransactionStatus getNextStatus(
+      JdbcSepTransaction txn, NotifyOnchainFundsSentRequest request)
+      throws InvalidRequestException {
+    return COMPLETED;
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/action/NotifyOnchainFundsSentHandler.java:61]: It's completed for withdrawal, pending_external for deposit

File: platform/src/main/java/org/stellar/anchor/platform/action/NotifyOnchainFundsSentHandler.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  @Override
+  protected Set<SepTransactionStatus> getSupportedStatuses(JdbcSepTransaction txn) {
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/action/NotifyOnchainFundsSentHandler.java:65]: withdrawal -> pending_anchor
deposit -> pending_user_transfer_start

File: platform/src/main/java/org/stellar/anchor/platform/utils/RpcUtil.java [hunk 3]
```diff
@@ -16,7 +16,11 @@ public class RpcUtil {
   public static final String JSON_RPC_VERSION = "2.0";
 
   public static RpcResponse getRpcSuccessResponse(Object id, Object result) {
-    return RpcResponse.builder().jsonrpc(JSON_RPC_VERSION).id(id).result(result).build();
+    return RpcResponse.builder()
+        .jsonrpc(JSON_RPC_VERSION)
+        .id(id)
+        .result(id != null ? result : null)
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/utils/RpcUtil.java:22]: BTW, I think id should be mandatory cc @JakeUrban

Comment (MazurakIhor) [reply platform/src/main/java/org/stellar/anchor/platform/utils/RpcUtil.java]: You are right. Recently Jake provided us details of how SorobanRPC is developed.
In this PR I made id field mandatory. In next PR we will make changes in response to include transaction id in error/result

File: platform/src/main/java/org/stellar/anchor/platform/action/DoStellarPaymentHandler.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+  @Override
+  protected void updateTransactionWithAction(
+      JdbcSepTransaction txn, DoStellarPaymentRequest request) throws AnchorException {
+    JdbcSep24Transaction txn24 = (JdbcSep24Transaction) txn;
+    if (horizon.isTrustLineConfigured(txn24.getToAccount(), txn24.getAmountOutAsset())) {
+      // TODO: Do we need to send request body?
+      custodyService.createTransactionPayment(txn24.getId(), null);
+    } else {
+      // TODO: Add account and asset to DB table, so that the cron job can check trust
```
[the remark this answers is not in the document]
Comment (MazurakIhor) [reply platform/src/main/java/org/stellar/anchor/platform/action/DoStellarPaymentHandler.java:99]: We will merge PR 987 just after this one.

File: platform/src/main/java/org/stellar/anchor/platform/action/NotifyOnchainFundsSentHandler.java (hunk 1 above)
Comment (MazurakIhor) [reply platform/src/main/java/org/stellar/anchor/platform/action/NotifyOnchainFundsSentHandler.java:61]: In development, we used new state diagrams that were presented to/approved by Jake:
Stellar_sep24_RPC-Sep24 - withdrawal v 0 4 drawio
Stellar_sep24_RPC-Sep24 - deposit v 0 4 drawio

File: platform/src/main/java/org/stellar/anchor/platform/action/NotifyOnchainFundsSentHandler.java (hunk 2 above)
Comment (MazurakIhor) [reply platform/src/main/java/org/stellar/anchor/platform/action/NotifyOnchainFundsSentHandler.java:65]: The same as above

**Review by Ifropc — APPROVED** [review]

**Pull request merged**